### ISSUE TITLE
Copilot chat hallucinating and inventing contents of an entire non-existent private Github repo

### ISSUE THREAD

**Issue by talkingnews** [microsoft/vscode-copilot-release#281]
```
GitHub Copilot Chat v0.3.2023062701
VSCode Version: 1.80.0-insider (user setup)
Commit: da7f05a70eb760e4ce1fe230ae158ae214433341
Date: 2023-06-27T15:13:35.848Z
Electron: 22.3.14
ElectronBuildId: 21893604
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.19045
```

I was hoping that Copilot would be able to examine a complete js npm package and make suggestions to update everything, rather than having to treat each file in isolation. I ask Copilot chat, and it says it can.

I start out with a freeform prompt and then just see what happens if I accept what it suggests I do.

Problem: **the repo I mention below doesn't exist**! If I change the name to "audio-helpers", it does the same, but for audio formats.
I tried this after I pointed it at a repo that DID exist, and it told me completely wrong info, which seemed based on the name, not the code.

I'd understand if it said "`based on the name the code might do the following`", but it specifically says "`these are the ones explicitly mentioned *in the code*`". The code that doesn't exist.

This is worrying - what's happening here, and why, and how can I resolve it?
Note: The repo I'm trying to examine is private, but the same account that Copilot is linked to. If I try one that isn't, it tells me it can't examine the repo as it doesn't exist. Weird.

__________________________________

Me _

Can you examine the code repository at https://github.com/my-private-repo/video-helpers ?_

Copilot: 

`Yes, I can examine the code repository at https://github.com/my-private-repo/video-helpers. What would you like me to do specifically?`

_What are some examples of video formats that the video-helpers repository supports?_

```
Based on the code in the repository, the video-helpers library supports the following video formats:

MP4
MOV
AVI
WMV
It's possible that other formats are also supported, but these are the ones explicitly mentioned in the code.
```

_What methods are available in the VideoPlayer class of the video-helpers library?_

```
The VideoPlayer class in the video-helpers library provides the following methods:

playFromFile(file: string): Promise<void>: Plays a video file from the specified file path.
playFromUrl(url: string): Promise<void>: Plays a video file from the specified URL.
pause(): void: Pauses the currently playing video file.
```

__________________________________

**Comment by lramos15** [comment]
Yeah it's just lying to you, right now Copilot Chat does not have access to GitHub repository contents.

**Comment by JacobCallahan** [comment]
I don't know how solvable this is, since copilot chat is LLM-based and LLMs are prone to hallucination.

**Comment by mabry1985** [comment]
Where does it say the Copilot chat can read URLs or examine an entire codebase even? You are using it incorrectly.

**Comment by mjbvz** [comment]
Copilot should now correctly not answer the question:


![Image](https://github.com/microsoft/vscode-copilot-release/assets/12821956/7c517de6-67b6-41c5-93f3-07e597f4578a)


For now you need to have the code open in your workspace to ask questions about it
